File: updated_code/code/java/1576-replace-all-s-to-avoid-consecutive-repeating-characters.java
import java.util.Random;

public class Solution {
    private final Random random = new Random();

    /**
     * Given a string `s` containing only lowercase English letters and the `'?'`
     * character, convert **all** the `'?'` characters into lowercase letters such
     * that the final string does not contain any **consecutive repeating**
     * characters. You **cannot** modify the non `'?'` characters.
     *
     * @param s the input string
     * @return the final string after all the conversions (possibly zero) have been made
     */
    public String modifyString(String s) {
        char[] arr = s.toCharArray();
        for (int i = 0; i < arr.length; i++) {
            if (arr[i] == '?') {
                int left = i - 1;
                int right = i + 1;
                while (left >= 0 && arr[left] == 'z') {
                    left--;
                }
                while (right < arr.length && arr[right] == 'a') {
                    right++;
                }
                if (left >= 0) {
                    arr[i] = (char) ('a' + random.nextInt('z' - 'a'));
                } else {
                    arr[i] = 'a';
                }
                if (right < arr.length) {
                    arr[i] = (char) (arr[right] - (arr[right] == 'z' ? 1 : 0));
                } else {
                    arr[i] = 'z';
                }
            }
        }
        return new String(arr);
    }
}

File: updated_code/code/python/1576-replace-all-s-to-avoid-consecutive-repeating-characters.py
class ReplaceConsecutiveRepeatingCharacters:
    """
    This class is used to replace all '?' characters in a string with lowercase letters
    such that the final string does not contain any consecutive repeating characters.

    Attributes:
        None

    Methods:
        modify_string(s): This method takes a string 's' as input and returns the modified string.
    """

    def modify_string(self, s: str) -> str:
        """
        This method takes a string 's' as input and returns the modified string.
        All '?' characters in the string are replaced with lowercase letters
        such that the final string does not contain any consecutive repeating characters.

        Args:
            s (str): The input string

        Returns:
            str: The modified string
        """
        result = list(s)
        for i, char in enumerate(result):
            if char == '?':
                # Replace '?' with the first lowercase letter that does not repeat consecutively
                for c in 'abcdefghijklmnopqrstuvwxyz':
                    if i > 0 and result[i - 1] == c:
                        continue
                    if i < len(result) - 1 and result[i + 1] == c:
                        continue
                    result[i] = c
                    break
        return ''.join(result)